src/utils/date.ts

```diff
@@ -0,0 +1,42 @@
+import { differenceInCalendarWeeks, getWeek, subMinutes, addMinutes } from 'date-fns';
+import { getTimezoneOffset } from 'date-fns-tz';
+
+/**
+ * The time data CTU provides is generated in the Europe/Prague timezone (GMT +1).
+ */
+export const CTU_TIMEZONE_OFFSET = getTimezoneOffset('Europe/Prague') / 3600000;
+
+/**
+ * Get "timezone corrected", date object.
+ */
+export function getOffsetDate() {
+  const date = new Date();
+  // how many minutes are we off from the CTU timezone
+  const offset = date.getTimezoneOffset() + CTU_TIMEZONE_OFFSET * 60;
+
+  if (offset === 0)
+    return date; // no offset, return the original date
+
+  // add or subtract the offset to get the correct date
+  return offset > 0 ?
+    addMinutes(date, Math.abs(offset)) :
+    subMinutes(date, Math.abs(offset));
+}
+
+/**
+ * Get the current week number (since the given semester start).
+ * @param semesterStartTimestamp unix timestamp of the semester start (in milliseconds).
+ */
+export function getSemesterWeekNo(semesterStartTimestamp: number) {
+  return 1 + differenceInCalendarWeeks(
+    getOffsetDate(),
+    new Date(semesterStartTimestamp),
+    { weekStartsOn: 1 },
+  );
+}
+
+export function getEvenOddWeek() {
+  return getWeek(
+    getOffsetDate(), { weekStartsOn: 1 }
+  ) % 2 === 1 ? 'even' : 'odd';
+}
```

Comment:
This could be improved to better utilize the `date-fns-tz` library:
```ts
import { differenceInCalendarWeeks, getWeek } from 'date-fns';
import { utcToZonedTime } from 'date-fns-tz';

const CTU_TIMEZONE = 'Europe/Prague';

/**
 * Get "timezone corrected" (for CTU), date object.
 */
export function getCTUCurrentDate() {
  return utcToZonedTime(new Date(), CTU_TIMEZONE);
}

/**
 * Get the current week number (since the given semester start).
 * @param semesterStartTimestamp unix timestamp of the semester start (in milliseconds).
 */
export function getSemesterWeekNo(semesterStartTimestamp: number) {
  return (
    1 +
    differenceInCalendarWeeks(
      getCTUCurrentDate(),
      new Date(semesterStartTimestamp),
      { weekStartsOn: 1 },
    )
  );
}

/**
 * Function returning appropriate string (odd/even) for the current week
 * (Uses "timezone corrected" (for CTU), date object)
 */
export function getEvenOddWeek() {
  return getWeek(getCTUCurrentDate(), { weekStartsOn: 1 }) % 2 === 1
    ? 'even'
    : 'odd';
}

```